tests/callbacks/test_Manager.cpp

```diff
@@ -0,0 +1,152 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/impl/type_traits.hpp"
+
+#include "kokkos-utils/callbacks/Events.hpp"
+#include "kokkos-utils/callbacks/Manager.hpp"
+
+#include "tests/callbacks/TestWorkload.hpp"
+
+/**
+ * @addtogroup unittests
+ *
+ * @c Kokkos callback manager
+ * --------------------------
+ *
+ * This group of tests check the behavior of @ref Kokkos::utils::callbacks::Manager.
+ */
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+namespace Kokkos::utils::tests::callbacks
+{
+using namespace Kokkos::utils::callbacks;
+
+//! @test Check properties of @ref Kokkos::utils::callbacks::Manager being a singleton class.
+TEST(Manager, singleton_traits)
+{
+    static_assert( ! std::is_copy_constructible_v<Manager>);
+    static_assert( ! std::is_copy_assignable_v   <Manager>);
+
+    static_assert( ! std::is_move_constructible_v<Manager>);
+    static_assert( ! std::is_move_assignable_v   <Manager>);
+}
+
+template <Event... EventTypes>
+struct TesterListener
+{
+    template <Kokkos::utils::impl::IsTypeOneOf<EventTypes...> EventType>
```

Comment:
It seems that this is some sort of stack ? (each event type has its own stack ?)

Worth documenting briefly.

Reply:
OK! Clearer indeed.